lib/searchkick.rb

```diff
@@ -77,7 +77,25 @@ def self.new_client
       end
   end
 
+  def self.get_current_client_id
+    return nil unless Thread.current['search_kick_client_id'].present?
+    Thread.current['search_kick_client_id'].to_i
+  end
+
   def self.writing_clients
+    current_client = get_current_client_id
+    if current_client.present?
+      case current_client
+      when 1
+        return [client]
+      when 2
+        raise Exception, 'new_client not defined' if new_client.nil?
```

Comment:
I don't understand this code. why is there a return after raise? Also never user Exception, use StandardException.
Exception isn't caught by default in a begin rescue block

Reply:
We this this code backwards compatible

Reply:
the code is backwards compatible. the`current_client` variable should be nil in case we don't write to a specific client (as worked before). the exception is thrown in case the second client is not defined (should never happen unless we are having issues with the client or if it was not defined for some reason - edge case). 

+ i will change to StandardException, wasn't aware to that